math/octave/Portfile

```diff
@@ -51,15 +53,18 @@ patchfiles-append patch-link.diff
 # allow mkoctfile to recognize and pass along -arch compiler switch
 patchfiles-append patch-arch_arg.diff
 
-# octave compiles in the hostname value
-# it is used to determine installation location
+# see https://savannah.gnu.org/bugs/?50025
+patchfiles-append patch-crash_on_exit.diff
+
+# octave compiles in the hostname value and its own full version number
+# they are used to determine installation locations
 #    of architecture-dependent package files
-# this means the build is dependent on the minor OS number
+# this means the build is dependent on the minor numbers of the OS and Octave
 # the following patch attempts to prevent this
-patchfiles-append patch-host_name.diff
+patchfiles-append patch-versions.diff
 
-# see https://savannah.gnu.org/patch/index.php?9439
-patchfiles-append patch-java.diff
+# reports of `__opengl_info__` not working the first time
+patchfiles-append patch-__opengl_info__.diff
 
 post-patch {
     if { ${os.arch} eq "i386" } {
```

Comment:
Don't know if it even matters, but I updated this line to:
```
    if { ${os.arch} eq "i386" || ${os.arch} eq "x86_64" } {
```

Reply:
I believe ``${os.arch}`` can only be ``powerpc`` or ``i386`` (see [here](https://github.com/macports/macports-base/blob/556a929a3917bd0a2c02166e916c2e4cc5f99172/src/macports1.0/macports.tcl#L673)).